internal/cli/sign_test.go

```diff
@@ -147,6 +169,147 @@ NumTCS=32
 	require.NoError(cli.Sign("exefile"))
 }
 
+func TestSignJSONExecutablePayload(t *testing.T) {
+	require := require.New(t)
+	assert := assert.New(t)
+
+	// Setup test environment
+	fs := afero.Afero{Fs: afero.NewMemMapFs()}
+	runner := signRunner{fs: fs}
+	cli := NewCli(&runner, fs)
+
+	// Copy from hostfs to memfs
+	copyBuffer, err := ioutil.ReadFile("../test/test_sign_elf_unsigned")
+	require.NoError(err)
+	err = fs.WriteFile("helloworld", copyBuffer, 0)
+	require.NoError(err)
+
+	// Check if no payload exists
+	unsignedExeMemfs, err := fs.Open("helloworld")
+	require.NoError(err)
+	unsignedExeMemfsStat, err := unsignedExeMemfs.Stat()
+	require.NoError(err)
+	unsignedExeMemfsSize := unsignedExeMemfsStat.Size()
+	payloadSize, payloadOffset, oeInfoOffset, err := getPayloadInformation(unsignedExeMemfs)
+	assert.NoError(err)
+	assert.Zero(payloadSize)
+	assert.Zero(payloadOffset)
+	assert.NotZero(oeInfoOffset)
+	unsignedExeMemfs.Close()
+
+	// Create a default config we want to check
+	testConf := &config{
+		Exe:             "helloworld",
+		Key:             defaultPrivKeyFilename,
+		Debug:           true,
+		HeapSize:        512, //[MB]
+		ProductID:       1,
+		SecurityVersion: 1,
+	}
+
+	// Marshal config
+	jsonData, err := json.Marshal(testConf)
+	require.NoError(err)
+	expectedLengthOfPayload := len(jsonData)
+
+	// Embed json config to helloworld
+	err = cli.embedConfigAsPayload("helloworld", jsonData)
+	assert.NoError(err)
+
+	// Check if new helloworld_signed now contains signed data
+	signedExeMemfs, err := fs.Open("helloworld")
+	require.NoError(err)
```

Comment:
```suggestion
	require.NoError(err)
	defer signedExeMemfs.Close()
```
instead of cleanup at the end